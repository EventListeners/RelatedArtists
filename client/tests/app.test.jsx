import React from 'react';
import App from '../src/app.jsx'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallow } from 'enzyme';
import 'jest-enzyme';


Enzyme.configure({ adapter: new Adapter() });


describe('test App ', () => {
  it('renders correctly', () => {
    const app = shallow(<App/>);
    expect(app).toMatchSnapshot();
  });

  it('should handle state changes', () => {
    const output = shallow(<App/>); 
    expect(output.state().clicked).toEqual(false);
    output.simulate('click');
    expect(output.state().clicked).toEqual(true);
  });

  it('should handle the click event', () => {
    handleClick = jest.fn();
    const output = shallow( <App/>);
    output.simulate('click');
    expect(handleClick).toHaveBeenCalledWith('clicked');
  });

});