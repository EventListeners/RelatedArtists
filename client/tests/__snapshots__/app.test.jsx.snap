// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App  renders correctly 1`] = `
<div>
  <div
    className="RAContainer"
  >
    <h1>
      Related Artists
    </h1>
    <div
      id="panel panel-default"
    >
      <div
        id="panel-body"
      >
        <relatedArtist
          relatedArtists={Array []}
        />
      </div>
      <div
        class="panel-footer"
        className="RAbutton"
      >
        <button
          onClick={[Function]}
        >
          More Artists
        </button>
      </div>
    </div>
  </div>
</div>
`;
